python/python/tests/test_async_client.py

```diff
@@ -853,6 +853,31 @@ async def test_lpushx(self, redis_client: TRedisClient):
         with pytest.raises(RequestError):
             await redis_client.lpushx(key1, [])
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_blpop(self, redis_client: TRedisClient):
+        key1 = f"{{test}}-1-f{get_random_string(10)}"
+        key2 = f"{{test}}-2-f{get_random_string(10)}"
+        value1 = "value1"
+        value2 = "value2"
+        value_list = [value1, value2]
+
+        assert await redis_client.lpush(key1, value_list) == 2
+        assert await redis_client.blpop([key1, key2], 0.5) == [key1, value2]
+
+        assert await redis_client.blpop(["non_existent_key"], 0.5) is None
```

Comment:
add a comment - .5 is higher than client request timeout
add a test with 0 timeout, but wrap it with something like `await asyncio.wait_for(future, timeout=5)` to avoid real blocking

Reply:
https://github.com/Bit-Quill/glide-for-redis/pull/209/files#diff-5db9a3bd41d647cfdba521ea248ea07d7c78bc9aaa41a26bb923d4f92a3efdd2R1691-R1699